src/aspire/source/image.py

```diff
@@ -272,18 +317,18 @@ def get_metadata(self, metadata_fields, indices=None, default_value=None):
 
         return result.to_numpy().squeeze()
 
-    def _images(self, start=0, num=np.inf, indices=None):
+    def _images_super(self, indices):
```

Comment:
So after playing around with this idea, I realized there's a tricky part. The `super()._images()` method contains two pieces of common code:

Before querying the subclass `_images` method it checks for `cached_im`:

```python
if self._cached_im is not None:
            logger.info("Loading images from cache")
            im = Image(self._cached_im[indices, :, :])
```

But then after getting `im = self._images(indices)`, it then applies the transformations to that **image** object

```python
        else:
            im = self._images(indices)
        return self.generation_pipeline.forward(im, indices)
```
We need the `Image` object in order to apply the transforms, but we don't want to generate the `Image` before the `cached_im` check..

I guess we could break this out into two functions, one of which checks for cached, and the other which applies transforms. We could call the second one `_images` and make it an abstract method still. What do you think?

Reply:
We discussed trying two approaches here, one with some mild code duplication and another using some on entry and exit helper functions (like a decorator).  Since we're currently sort of backed up, I'm thinking we just move forward with the code duplication so we can move on to more interesting things.